networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -212,64 +213,80 @@ def _add_floating_ip(self, ri, ex_gw_port, floating_ip, fixed_ip):
         self._asr_do_add_floating_ip(ri, floating_ip, fixed_ip,
                                      vrf_name, ex_gw_port)
 
+        # We need to make sure that our external interface has an IP address
+        # on the same subnet as the floating IP (needed in order to handle ARPs
+        # on the external interface). Search for the matching subnet for this
+        # FIP, and use the highest host address as a secondary address on that
+        # interface
+        subnets = ri.router['gw_port'].get('extra_subnets', [])
+        subnet = self._get_matching_subnet(subnets, floating_ip)
+        if subnet:
+            secondary_ip = netaddr.IPAddress(subnet.value +
+                                             (subnet.hostmask.value - 1))
+            if secondary_ip:
+                self._asr_do_add_secondary_ip(secondary_ip,
```

Comment:
Yeah -- it will reconfigure the same secondary IP. We could do an optimization here, but I was holding off on that until we added per-VRF allocations. I've been using this, and it hasn't caused any problems during system testing.